### ISSUE TITLE
Xamarin.Mac x64 2.0 vs MonoMac x64

### ISSUE THREAD

**Issue by Krakean** [picoe/Eto#989]
Hello.
Curious, which one project is better to use for Mac - Xamarin-based or Mono-based?
Is there some principal differences, like usability/stability and/or performance?
Enlight me please, because I currently working on my game engine, and would like to use Eto as cross-platform UI solution for my engine's editor (for Win-part I would like to use Eto.Wpf as it seems to allow implement custom controls easily, but in case of Mac I curious which one is right to use)

Thanks for help

**Comment by cwensley** [comment]
Hi @Krakean, Xamarin.Mac (or Eto.XamMac2) is the preferred platform, if you can use it.  First, it allows you to bundle mono in with your app, and second it has all the newest API's in the mac platform if you're doing anything platform specific.

Typically, MonoMac (Eto.Platform.Mac nuget) was used so you can create an .app bundle even without a Mac, but in 2.4.0 the Eto.Platform.XamMac2 nuget package will also be able to do the same thing.  This however is experimental and I don't know if there will be any problems, hence the Mac64 platform.  Either way, creating an .app bundle on non-mac will always require Mono to be installed on Mac to run the app.

MonoMac will probably also be used once I get around to making a .NET Core compatible package so you can bundle .NET Core instead of using the mono runtime (unless Xamarin folk make Xamarin.Mac work on it).  Theoretically you'd be able to package this on any platform and run without any dependencies on Mac.

So to sum up: use Xamarin.Mac if you can.

Hope this helps!
Curtis.

**Comment by Krakean** [comment]
@cwensley 
Thank you very much for answer.
Please, keep your good work on Eto! :)

**Comment by Krakean** [comment]
@cwensley 
Two more questions:
1. what's difference between XamMac & XamMac2-modern csprojs if to speak about Eto.Test.Mac?
2. there is no .NET Core 2.0 support for Eto.Test (+.Mac/.Wpf) yet, right?

**Comment by cwensley** [comment]
Hey @Krakean,

This is a rundown of the differences between Xamarin.Mac variations:

XamMac: Xamarin.Mac classic (similar to MonoMac, discouraged to use)
XamMac2-net45: Xamarin.Mac "unified" full or .net 4.5+
XamMac2-mobile: Xmarin.Mac "unified" mobile profile (similar to .net standard 2.0)

It really depends on what other libraries you need, but the mobile profile is the preferred if you can use it as Xamarin.Mac can strip more from the framework and make it smaller even with mono bundled with your app.

There is no .NET Core 2.0 for Mac or Windows yet.  WPF will never get .NET Core support as WPF (and WinForms) require the full .NET framework.  However we should be able to get Mac support using .NET Core, we just need to get things built for it (monomac for .net standard, which is mostly already done), the tooling (VS on Mac) updated for debugging, and create nuget packages to make it easy to build/package the .app bundle.
